streaming/base/format/base/writer.py

```diff
@@ -294,12 +269,13 @@ class JointWriter(Writer):
     """Writes a streaming dataset with joint shards.
 
     Args:
-        local: (str, optional): Optional local output dataset directory. If not provided, a random
-           temp directory will be used. If ``remote`` is provided, this is where shards are cached
-            before uploading. One or both of ``local`` and ``remote`` must be provided. Defaults to
-            ``None``.
-        remote: (str, optional): Optional remote output dataset directory. If not provided, no
-            uploading will be done. Defaults to ``None``.
+        out (str | List[str]): Output dataset directory to save shard files.
```

Comment:
this should probably be `Union[str, Tuple[str, str]]` due to fixed length

Reply:
Thanks. Addressed in https://github.com/mosaicml/streaming/pull/171/commits/b6539a43b3035c50d86be6554786296bdc1d6df4.